app/services/submit_application.rb

```diff
@@ -53,6 +53,18 @@ def email_address_is_a_bot?(reference)
   def submit_application
     application_choices.each do |application_choice|
       SubmitApplicationChoice.new(application_choice).call
+
+      if application_form.apply_again? && enough_references_have_been_provided?
```

Comment:
I think we could skip checking `apply_again?` here, as it makes sense without it too.

Reply:
You're right I think. I guess I was thinking of limiting the extent of the change but one less special case is better in the long run. But I do think this should strictly speaking be guarded by the feature flag?